### PR TITLE
refactor: review the `argilla. _constants` module usage

### LINKED ISSUE #4469: [TASK ITEM] Review `_constans` module usage

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

In this PR the module `argilla._constants` has been reviewed to separate server-side and client-side constant definitions. The server module contains its constant definitions, even if they're duplicated. This will help to separate client and server modules into different packages.

Closes #4469 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactor (change restructuring the codebase without changing functionality)
- [X] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

Changes have been tested locally. 

**Checklist**

- [ ] I added relevant documentation
- [X] I followed the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

**Review by gabrielmbmb — APPROVED** [review]

File: src/argilla/server/constants.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+DEFAULT_MAX_KEYWORD_LENGTH = 128
+DEFAULT_TELEMETRY_KEY = "C6FkcaoCbt78rACAgvyBxGBcMB3dM3nn"
+
+
+# The metadata field name prefix defined for protected (non-searchable) values
+PROTECTED_METADATA_FIELD_PREFIX = "_"
+
+ES_INDEX_REGEX_PATTERN = r"^(?!-|_)[a-z0-9-_]+$"
+
+_JS_MAX_SAFE_INTEGER = 9007199254740991
```
Comment (jfcalvo) [inline src/argilla/server/constants.py]: We're using it outside this module so I guess we can remove the underscore:

```suggestion
JS_MAX_SAFE_INTEGER = 9007199254740991
```

File: src/argilla/server/constants.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+API_KEY_HEADER_NAME = "X-Argilla-Api-Key"
+WORKSPACE_HEADER_NAME = "X-Argilla-Workspace"
+
+DEFAULT_USERNAME = "argilla"
+DEFAULT_PASSWORD = "1234"
+DEFAULT_API_KEY = "argilla.apikey"
+
+DEFAULT_MAX_KEYWORD_LENGTH = 128
+DEFAULT_TELEMETRY_KEY = "C6FkcaoCbt78rACAgvyBxGBcMB3dM3nn"
+
```
Comment (jfcalvo) [inline src/argilla/server/constants.py:24]:
```suggestion
```

**Review by jfcalvo — APPROVED** [review]
Looks fine to me. I added some small suggestions in the case that you think it's ok to apply them.

File: src/argilla/server/constants.py (hunk 2 above)
Comment (frascuchon) [reply src/argilla/server/constants.py:24]: Done

**Pull request merged**